HerPublicWebsite.UnitTests/BusinessLogic/Services/PolicyTeamUpdateServiceTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using FluentAssertions.Extensions;
+using HerPublicWebsite.BusinessLogic;
+using HerPublicWebsite.BusinessLogic.ExternalServices.EmailSending;
+using HerPublicWebsite.BusinessLogic.Models;
+using HerPublicWebsite.BusinessLogic.Services.CsvFileCreator;
+using HerPublicWebsite.BusinessLogic.Services.PolicyTeamUpdate;
+using HerPublicWebsite.BusinessLogic.Services.RegularJobs;
+using Moq;
+using NUnit.Framework;
+using Tests.Builders;
+
+namespace Tests.BusinessLogic.Services;
+
+public class PolicyTeamUpdateServiceTests
+{
+    private Mock<IDataAccessProvider> mockDataProvider;
+    private Mock<ICsvFileCreator> mockCsvFileCreator;
+    private Mock<IWorkingDayHelperService> mockWorkingDayHelperService;
+    private Mock<IEmailSender> mockEmailSender;
+    private Mock<IReferralFilterService> mockReferralFilterService;
+    private PolicyTeamUpdateService policyTeamUpdateService;
+
+    [SetUp]
+    public void Setup()
+    {
+        mockDataProvider = new Mock<IDataAccessProvider>();
+        mockCsvFileCreator = new Mock<ICsvFileCreator>();
+        mockWorkingDayHelperService = new Mock<IWorkingDayHelperService>();
+        mockEmailSender = new Mock<IEmailSender>();
+        mockReferralFilterService = new Mock<IReferralFilterService>();
+        policyTeamUpdateService = new PolicyTeamUpdateService(
+            mockDataProvider.Object,
+            mockCsvFileCreator.Object,
+            mockWorkingDayHelperService.Object,
+            mockEmailSender.Object,
+            mockReferralFilterService.Object);
+    }
+
+    [Test]
+    public async Task SendPolicyUpdate_WhenCalled_CallsCsvFileCreatorWithReferralsIncludedInFilter()
+    {
+        // Arrange
+        var validReferral = new ReferralRequestBuilder(1)
+            .Build();
+        var invalidReferral = new ReferralRequestBuilder(2)
+            .Build();
+        
+        var allReferrals = new List<ReferralRequest>()
+        {
+            validReferral, invalidReferral
+        };
+        var filteredReferrals = new List<ReferralRequest>()
+        {
+            validReferral
+        };
+
+        mockDataProvider
+            .Setup(dp =>
+                dp.GetReferralRequestsBetweenDates(It.IsAny<DateTime>(), It.IsAny<DateTime>()))
+            .ReturnsAsync(allReferrals);
+
+        mockDataProvider
+            .Setup(dp =>
+                dp.GetAllReferralRequests())
+            .ReturnsAsync(allReferrals);
+
+        mockReferralFilterService
+            .Setup(rfs =>
+                rfs.FilterForSentToNonPending(allReferrals))
+            .Returns(filteredReferrals);
```

Comment:
Can we use the real `ReferralFilterService` rather than a mocked one? That's the bit we really want to test.

Reply:
I did consider this, though the tests for ReferralFilterService already enforce this logic. To me, the only part these tests need be concerned with is that the filter is obeyed

Reply:
At the moment, all these tests are doing is saying "we call this function and use the result" - if that function were to be updated to do something else then this would still be passing. It's not that both sets of tests are totally without value, but at the moment they don't tell us anything that wouldn't be clear from looking at the code: in this case that things are passed between functions correctly (this is what happens when you mock everything), and for the `ReferralFilterService` that the single where clause is doing what it's supposed to do.

If on the other hand this test uses the real service, then if someone comes along and changes the service in a way that would change the contents of these CSVs, the test will fail and they'll know about it. The test would encode the actual functionality, rather than being tied to the implementation. Put another way: the 'unit' here that we want to test is the piece of functionality that says "only referrals not submitted to pending LAs are included", not the bit of code you've written.

(you could go further and argue that the 'unit' extends to the methods generating the CSV from the referrals, and so our tests should be more like "given these referrals exist, this is the CSV I get back", but I think there's enough specific business logic in the CSV generation that it warrants its own tests covering it, and we don't want to cover the same thing twice)